cmd/virt-launcher/virt-launcher.go

```diff
@@ -496,6 +513,12 @@ func main() {
 			finalShutdownCallback,
 			gracefulShutdownCallback)
 
+		ctx, cancelCtx := context.WithCancel(context.Background())
+		defer cancelCtx()
+
+		if err := runDownwardMetricsVirtioServer(ctx); err != nil {
```

Comment:
maybe we need to explicitly pass the `signalStopChan`?  so we will close this server properly. What do you think?

Reply:
@fossedihelm could you please also take a look?

Reply:
Do you mean?
```go
if err := runDownwardMetricsVirtioServer(ctx, signalStopChan); err != nil
``

Reply:
Done.